compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/repositories/RemotePackageRepository.java

```diff
@@ -242,27 +243,35 @@ private PackageNameResolutionRequest toPackageNameResolutionRequest(List<ImportM
         if (options.offline()) {
             return cachedPackages;
         }
-
-        // Resolve the requests from remote repository
-        try {
-            PackageResolutionRequest packageResolutionRequest = toPackageResolutionRequest(requests);
-            PackageResolutionResponse packageResolutionResponse = client.resolveDependencies(
-                    packageResolutionRequest, JvmTarget.JAVA_11.code(),
-                    RepoUtils.getBallerinaVersion(), true);
-
-            Collection<PackageMetadataResponse> remotePackages =
-                    fromPackageResolutionResponse(requests, packageResolutionResponse);
-            // Merge central requests and local requests
-            // Here we will pick the latest package from remote or local
-            return mergeResolution(remotePackages, cachedPackages);
-
-        } catch (ConnectionErrorException e) {
-            // ignore connect to remote repo failure
-            // TODO we need to add diagnostics for resolution errors
-        } catch (CentralClientException e) {
-            throw new ProjectException(e.getMessage());
+        // Remove the already resolved requests when the locking mode is hard
+        for (PackageMetadataResponse response : cachedPackages) {
+            if (response.packageLoadRequest().packageLockingMode().equals(PackageLockingMode.HARD)
+                    && response.resolutionStatus().equals(ResolutionResponse.ResolutionStatus.RESOLVED)) {
+                requests.remove(response.packageLoadRequest());
+            }
+        }
+        // Resolve the requests from remote repository if there are unresolved requests
+        if (requests.size() > 0) {
```

Comment:
```suggestion
        if (requests.isEmpty()) {
```

Reply:
@lnash94 it should be `if (!requests.isEmpty()) {`

Reply:
Fixed with 9113a96f8a8d84844aae0b5b4cb4e4aa61df0f50